### PR TITLE
NRT implementation

### LINKED ISSUE #181: Add Kinchin-Pease and NRT approximations
On larger length scales (many microns) we can stop full recoil simulation at a given energy cutoff and use Kinchin-Pease or NRT to set the defect numbers (and deposit the remaining energy into the current element).

### PR DESCRIPTION
**Pull request by snschune**
Closes #181

### CODE REVIEW

**Comment by snschune** [comment]
I think this is ready for review. The missing piece is using the derivative capability in the pattern matching and verification that the derivative is actually computed right.

**Comment by dschwen** [comment]
You should be able to fix your conflict using

```
git rebase --strategy-option="rename-threshold=10" upstream/devel
```

**Comment by snschune** [comment]
Failure is a simple timeout in debug, hence unrelated.

**Comment by snschune** [comment]
These tests are not reproducible in parallel. I vary the number densities in the domain so each processor sees a different number density range. The recomputation of NRT objects is done whenever the composition in an element is too far from the closest match. Depending on how you partition, different NRT objects will be computed. 

Restrict to 1 proc?

**Comment by dschwen** [comment]
For now yes. Let's discuss when I'm back.

File: examples/nrt_comparison/U_xZr_1-x_C/tests [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[Tests]
+  [./Ux_Zr_1-x_C]
+    type = 'RunApp'
+    input = 'parkin_coulter_UxZr1-xC.i'
+    check_input = True
```
Comment (dschwen) [inline examples/nrt_comparison/U_xZr_1-x_C/tests]: Add

```
required_objects = 'PolyatomicRecoil'
```

here and wherever applicable

File: include/utils/PolyatomicDisplacementFunctionBase.h [hunk 2]
```diff
@@ -0,0 +1,142 @@
+/**********************************************************************/
+/*                     DO NOT MODIFY THIS HEADER                      */
+/* MAGPIE - Mesoscale Atomistic Glue Program for Integrated Execution */
+/*                                                                    */
+/*            Copyright 2017 Battelle Energy Alliance, LLC            */
+/*                        ALL RIGHTS RESERVED                         */
+/**********************************************************************/
+
```
Comment (dschwen) [inline include/utils/PolyatomicDisplacementFunctionBase.h]: Add

```
#ifdef GSL_ENABLED
```

here and where applicable

File: src/userobjects/MyTRIMRasterizer.C [hunk 3]
```diff
@@ -531,6 +538,16 @@ MyTRIMRasterizer::periodicPoint(const Point & pos) const
   return p;
 }
 
+bool
+MyTRIMRasterizer::isTrackedSpecies(unsigned int atomic_number, Real mass_number) const
```
Comment (dschwen) [inline src/userobjects/MyTRIMRasterizer.C]: can we use ionTag here?

**Comment by dschwen** [comment]
You are using the workspace wrapper. See my changes in 768efb29ce10a09776499e204076ea128dd967c5 for how to rewrite this so it works with GSL 2.1

**Pull request closed without merging**